plugin/pkg/auth/authorizer/webhook/webhook.go

```diff
@@ -180,10 +185,10 @@ func (w *WebhookAuthorizer) Authorize(attr authorizer.Attributes) (err error) {
 		}
 	}
 	if r.Status.Allowed {
-		return nil
+		return true, "", nil
 	}
 	if r.Status.Reason != "" {
-		return errors.New(r.Status.Reason)
+		return false, r.Status.Reason, nil
 	}
-	return errors.New("unauthorized")
+	return false, "", nil
```

Comment:
I think this whole section can reduce to this (just pass through the Allowed and Reason fields):
```
return r.Status.Allowed, r.Status.Reason, nil
```